### PR TITLE
Update robotics course tutorials

### PR DESCRIPTION
**Pull request by SissiFeng**
Added quiz and assignment links, added a little more explanatory words.

### CODE REVIEW

**Comment by sgbaird** [comment]
@linx5o @Luthiraa @RubainaFarin31 can you have a look at the file changes and provide some feedback?

See https://github.com/AccelerationConsortium/ac-microcourses/issues/71

File: docs/courses/robotics/3.1-pumps-and-pipettes.md [hunk 1]
```diff
@@ -119,6 +157,12 @@ set_position(1.9)  # Almost full extension
 pwm.deinit()  # Deinitialize PWM
 ```
 
+#### Cool Uses 😎
+
+- 🍫 Chocolatiers use them for adding precise amounts of flavoring
```
Comment (sgbaird) [inline docs/courses/robotics/3.1-pumps-and-pipettes.md:162]: Some of these seem a bit ChatGPT-like. Is this something we trust? If not, maybe remove this line

File: docs/courses/robotics/3.1-pumps-and-pipettes.md [hunk 2]
```diff
@@ -87,12 +114,23 @@ Based on https://github.com/ac-rad/digital-pipette
 - [Wire cutter/stripper combo](https://www.amazon.ca/Professional-crimping-Multi-Tool-Stripper-Multi-Function/dp/B073YG65N2)
 - [Mini flathead screwdriver set](https://www.amazon.ca/dp/B08QCT9NHY/) (one for regulators and a smaller one for PiCowbell terminals)
 
+Digital pipettes are the unsung heroes of precise liquid handling! 🦸‍♀️ They're essential in:
```
Comment (sgbaird) [inline docs/courses/robotics/3.1-pumps-and-pipettes.md:117]: "Digital Pipette" is actually somewhat specific to the design by Naruki and co. "Automated single-channel syringe pump" is probably the best generic term for Naruki's "Digital Pipette".

@n-yoshikawa feel free to weigh in if you have a moment.

File: docs/courses/robotics/3.1-pumps-and-pipettes.md [hunk 3]
```diff
@@ -177,8 +230,29 @@ tic.deenergize()
...
+
+- 🎰 Slot machines use them for spinning reels
+- 🎹 Electronic musical instruments for key action
+- 🚗 Some car headlights use them for adaptive lighting systems
+
 ### Fan Control
 
+Temperature control is crucial in many lab setups! 🌡️ Fan controllers help in:
+
+- 🧊 Maintaining precise temperatures in incubators
```
Comment (sgbaird) [inline docs/courses/robotics/3.1-pumps-and-pipettes.md:243]: Let's also mention that many custom lab setups use magnets attached to fans to control stir bars for chemistry experiments, e.g., [pioreactor](https://pioreactor.com/).

File: docs/courses/robotics/3.2-serial-communication.md [hunk 4]
```diff
@@ -10,6 +10,27 @@ Design and execute software to read data from a mass balance using serial commun
 
 ## 🔰 Tutorial
 
+Serial communication is a method of transmitting data between devices one bit at a time. It's widely used in scientific instruments, industrial equipment, and embedded systems due to its simplicity and reliability. In this tutorial, we'll focus on RS232, a common serial communication standard, and how to interface it with modern microcontrollers.
```
Comment (sgbaird) [inline docs/courses/robotics/3.2-serial-communication.md:13]: Maybe briefly clarify what "serial" means in the context of serial communication

File: docs/courses/robotics/3.2-serial-communication.md [hunk 5]
```diff
@@ -10,6 +10,27 @@ Design and execute software to read data from a mass balance using serial commun
...
+
+1. **Raspberry Pi Pico WH** 🥧
+   This little powerhouse is our brain! It's a versatile microcontroller with built-in UART capabilities, making it perfect for serial communication projects.
+
+   Fun fact: The 'WH' stands for 'with headers' - it comes pre-soldered for easy use! 🔌
+
+2. **Analytical Balance** ⚖️
+   We're using the A&D HR-100A, a high-precision scale that can measure up to 102g with an accuracy of 0.0001g! 
+
+   Did you know? This balance is so sensitive it can detect the weight of a fingerprint! 👆💨
```
Comment (sgbaird) [inline docs/courses/robotics/3.2-serial-communication.md:27]: Maybe clarify in parantheses (~0.0002 mg)

File: docs/courses/robotics/3.2-serial-communication.md [hunk 6]
```diff
@@ -20,17 +41,44 @@ Design and execute software to read data from a mass balance using serial commun
 - [Electrical wire (3 feet, black)](https://www.digikey.ca/en/products/detail/cnc-tech/10981-18-1-2000-001-1-TD/17799168)
 - [Mini flathead screwdriver set](https://www.amazon.ca/dp/B08QCT9NHY/) (one for DC Motor driver terminals and a smaller one for PiCowbell terminals)
 
+The Raspberry Pi Pico WH is a versatile microcontroller that's ideal for this project due to its built-in UART (Universal Asynchronous Receiver/Transmitter) capabilities. The RS232 to TTL converter is crucial as it translates the voltage levels between the RS232 standard (used by many scientific instruments) and the 3.3V logic used by the Pico.
```
Comment (sgbaird) [inline docs/courses/robotics/3.2-serial-communication.md:44]: Maybe mention that while UART to RS232 isn't a perfect solution, it's a convenient way to use the Pico W and should work fine for our setup

File: docs/courses/robotics/3.3-liquid-handlers.md [hunk 7]
```diff
@@ -7,11 +7,59 @@
...
 
-In this module, you will perform liquid transfer between containers with an automated liquid handler (Science Jubilee and Opentrons OT-2).
+Welcome to the fascinating world of automated liquid handling! 🌊 In this module, you'll learn how to perform liquid transfers between containers using two cutting-edge systems: the Science Jubilee and the Opentrons OT-2. 
+
+### The Liquid Handling Superstars 🌟
+
+1. **Science Jubilee** 🔬
+   Imagine a Swiss Army knife for scientists - that's Science Jubilee! It's a versatile, open-source platform that can handle various lab tasks, including precise liquid transfers.
+
+   Fun fact: Science Jubilee is like a scientific chameleon, adapting to different research needs! 🦎
```
Comment (sgbaird) [inline docs/courses/robotics/3.3-liquid-handlers.md:17]: Let's clarify that it's an open source toolchanger at heart. You can refer to the science jubilee docs for a definition (and we should probably link to these docs anyway https://science-jubilee.readthedocs.io/)

File: docs/courses/robotics/3.3-liquid-handlers.md [hunk 8]
```diff
@@ -7,11 +7,59 @@
...
+4. **Safety** 🛡️
+   They can handle hazardous materials, keeping human researchers safe.
+
+### Real-World Applications 🌍
+
+1. **Drug Discovery** 💊
+   Automated liquid handlers are crucial in high-throughput screening of potential drug compounds.
+
+2. **Genomics Research** 🧬
+   They're indispensable for tasks like DNA extraction and PCR setup.
```
Comment (sgbaird) [inline docs/courses/robotics/3.3-liquid-handlers.md:44]: Probably better to use softer language like "well-suited for" or "often used for", etc.

File: docs/courses/robotics/3.3-liquid-handlers.md (hunk 7 above)
Comment (sgbaird) [reply docs/courses/robotics/3.3-liquid-handlers.md:17]: EDIT: I'm also seeing the more detailed section below

Comment (sgbaird) [reply docs/courses/robotics/3.3-liquid-handlers.md:17]: Maybe for the fun fact, you can point to the research article where it was used for quantum dot synthesis: https://pubs.rsc.org/en/content/articlelanding/2023/dd/d3dd00033h

**Review by sgbaird — COMMENTED** [review]
Overall nice! Good to think about content from ChatGPT that might not be verified, and best to be a bit cautious when talking about the benefits of the various platforms (i.e., avoid grand statements)

File: docs/courses/robotics/3.2-serial-communication.md (hunk 5 above)
Comment (linx5o) [reply docs/courses/robotics/3.2-serial-communication.md:27]: Should the clarification be (~0.1mg) or (~0.0001g) instead?

**Review by linx5o — COMMENTED** [review]
The updates make the course more enjoyable and provides more context about each section.

I do still think that the demos could do with a bit more details and guides. Potentially a small task to accomplish with each piece of equipment.

**Comment by sgbaird** [comment]
I'm going to merge for now. Do you have an example or two in mind for something small like you mentioned?

**Pull request merged**